service/src/main/java/uk/gov/hmcts/reform/fpl/service/ManageDocumentService.java

```diff
@@ -0,0 +1,232 @@
+        Map<String, Object> listAndLabel = new HashMap<>();
+
+        UUID selectedC2DocumentCode = getDynamicListValueCode(caseData.getManageDocumentsSupportingC2List(), mapper);
+        List<Element<C2DocumentBundle>> c2DocumentBundle = caseData.getC2DocumentBundle();
+
+        IntStream.range(0, c2DocumentBundle.size())
+            .filter(index -> c2DocumentBundle.get(index).getId().equals(selectedC2DocumentCode))
+            .findFirst()
+            .ifPresent(index -> listAndLabel.put(SUPPORTING_C2_LABEL,
+                c2DocumentBundle.get(index).getValue().toLabel(index + 1)));
+
+        listAndLabel.put(SUPPORTING_C2_LIST_KEY, caseData.buildC2DocumentDynamicList(selectedC2DocumentCode));
+
+        return listAndLabel;
+    }
+
+    public List<Element<SupportingEvidenceBundle>> getFurtherEvidenceCollection(CaseData caseData) {
+        if (caseData.getManageDocument().isDocumentRelatedToHearing()) {
+            List<Element<HearingFurtherEvidenceBundle>> bundles = caseData.getHearingFurtherEvidenceDocuments();
+            if (!bundles.isEmpty()) {
+                UUID selectedHearingCode = getDynamicListValueCode(caseData.getManageDocumentsHearingList(), mapper);
+
+                Optional<Element<HearingFurtherEvidenceBundle>> bundle = findElement(
+                    selectedHearingCode, bundles
+                );
+
+                if (bundle.isPresent()) {
+                    return bundle.get().getValue().getSupportingEvidenceBundle();
+                }
+            }
+        } else if (caseData.getFurtherEvidenceDocuments() != null) {
+            return caseData.getFurtherEvidenceDocuments();
+        }
+
+        return getSupportingEvidenceBundle(null);
+    }
+
+    public List<Element<SupportingEvidenceBundle>> getC2SupportingEvidenceBundle(CaseData caseData) {
+        UUID selectedC2 = getDynamicListValueCode(caseData.getManageDocumentsSupportingC2List(), mapper);
+        C2DocumentBundle c2DocumentBundle = caseData.getC2DocumentBundleByUUID(selectedC2);
+
+        if (c2DocumentBundle.getSupportingEvidenceBundle() != null) {
+            return c2DocumentBundle.getSupportingEvidenceBundle();
+        }
+
+        return getSupportingEvidenceBundle(null);
+    }
+
+    public List<Element<SupportingEvidenceBundle>> getSupportingEvidenceBundle(
+        List<Element<SupportingEvidenceBundle>> supportingEvidenceBundleList) {
+        if (supportingEvidenceBundleList == null || supportingEvidenceBundleList.isEmpty()) {
+            return List.of(element(SupportingEvidenceBundle.builder().build()));
+        }
+
+        return supportingEvidenceBundleList;
+    }
+
+    public List<Element<HearingFurtherEvidenceBundle>> buildHearingFurtherEvidenceCollection(
+        CaseData caseData, List<Element<SupportingEvidenceBundle>> supportingEvidenceBundle) {
+
+        List<Element<HearingFurtherEvidenceBundle>> hearingFurtherEvidenceBundle;
```

Comment:
Not required split - the declaration on a separate line.

Reply:
done